File: samples/java_demo/src/com/example/service/NotificationService.java
package com.example.service;

import com.example.util.LoggerUtil;

public class NotificationService {
    private final EmailGateway emailGateway = new EmailGateway();

    public void sendPaymentStatus(String userId, boolean paid) {
        String status = paid ? "PAID" : "FAILED";
        LoggerUtil.info("Sending payment status " + status + " for " + userId);
        emailGateway.deliver(userId, "Payment status: " + status);
    }
}

File: samples/python_demo/app/service/notification_service.py
from app.service.email_gateway import EmailGateway
from app.util.logger_util import LoggerUtil


class NotificationService:
    def __init__(self) -> None:
        self._email_gateway = EmailGateway()

    def send_payment_status(self, user_id: str, paid: bool) -> None:
        status = "PAID" if paid else "FAILED"
        LoggerUtil.info(f"Sending payment status {status} for {user_id}")
        self._email_gateway.deliver(user_id, f"Payment status: {status}")
